docs/protocol/off-chain/overview.mdx

```diff
@@ -0,0 +1,245 @@
+- All updates redistribute existing funds
+- Cannot add or remove funds from active sessions
+- Must close and recreate sessions to change total funds
+
+**Use Case**: Maintained for existing applications, not recommended for new development.
+
+### NitroRPC/0.4 (Current)
+
+**Features**:
+- Intent-based state updates: **OPERATE**, **DEPOSIT**, **WITHDRAW**
+- Add funds to active <Tooltip content="Off-chain channels built on top of payment channels, intended to be used by app developers to enable application-specific interactions and transactions without touching the blockchain">app sessions</Tooltip> (DEPOSIT)
+- Remove funds from active sessions (WITHDRAW)
+- Better error handling and validation
+- Enhanced security checks
+
+**Use Case**: All new implementations should use this version.
+
+---
+
+## Communication Architecture
+
+Nitro RPC operates over WebSocket connections, enabling bidirectional real-time communication between clients and <Tooltip content="A virtual ledger layer that provides a unified ledger (through Nitro RPC) and coordinates state channels (through Nitrolite), providing chain abstraction for developers and users">a clearnode</Tooltip>.
+
+```mermaid
+graph TB
+    A[Client Application] <-->|WebSocket Connection| B[Clearnode]
+    B <-->|Event Monitoring| C[Blockchain]
+    B <-->|State Management| D[Database]
+    
+    A -->|"1. RPC Request<br/>(signed)"| B
+    B -->|"2. Process & Validate"| D
+    B -->|"3. RPC Response<br/>(signed)"| A
+    
+    B -.->|"Async Notifications"| A
+    C -.->|"Blockchain Events"| B
+    
+    style A fill:#e1f5ff
+    style B fill:#ffe1f5
+    style C fill:#f5ffe1
+    style D fill:#fff5e1
+```
+
+### Connection Flow
+
+1. **Client Establishes WebSocket**: Open persistent connection to <Tooltip content="A virtual ledger layer that provides a unified ledger (through Nitro RPC) and coordinates state channels (through Nitrolite), providing chain abstraction for developers and users">a clearnode</Tooltip>
+2. **Authentication**: Complete 3-step auth flow (auth_request → auth_challenge → auth_verify)
+3. **RPC Communication**: Send requests, receive responses
+4. **Notifications**: Receive real-time updates (balance changes, <Tooltip content="A secure communication pathway between participants that locks funds in an on-chain smart contract while enabling off-chain state updates">channel</Tooltip> events)
+5. **Keep-Alive**: Periodic ping/pong to maintain connection
+
+---
+
+## Message Categories
+
+Nitro RPC methods are organized into functional categories:
+
+### 1. Authentication Methods
+
+Establish and manage authenticated sessions:
+- `auth_request` - Initiate authentication
+- `auth_challenge` - Receive challenge from <Tooltip content="A virtual ledger layer that provides a unified ledger (through Nitro RPC) and coordinates state channels (through Nitrolite), providing chain abstraction for developers and users">a clearnode</Tooltip>
```

Comment:
I'd say that in the current protocol implementation, when you submit a request with `auth_request` method, then on success you receive a response with `auth_challenge` method